File: Services/src/main/java/services/IServices.java
package services;

import model.domain.Agency;
import model.domain.Reservation;
import model.domain.Trip;

import java.sql.Time;
import java.util.Collection;

public interface IServices {
    void login(Agency agency, IObserver observer) throws ServiceException;
    void logout(Agency agency, IObserver observer) throws ServiceException;

    Collection<Agency> getAgencies() throws ServiceException;
    Collection<Trip> getTrips(String destination, Time startTime, Time endTime) throws ServiceException;
    Collection<Reservation> getReservations() throws ServiceException;
    void saveReservation(Reservation reservation) throws ServiceException;
}

File: Services/IServices.cs
using System;
using System.Collections.Generic;
using Model;

namespace Services
{
    public interface IServices
    {
        void Login(Agency agency, IObserver observer);
        void Logout(Agency agency, IObserver observer);

        IEnumerable<Agency> GetAgencies();
        IEnumerable<Trip> GetTrips(string destination, TimeSpan startTime, TimeSpan endTime);
        IEnumerable<Reservation> GetReservations();
        void SaveReservation(Reservation reservation);
    }
}
